packages/astro/src/runtime/server/transition.ts

```diff
@@ -65,6 +74,20 @@ export function renderTransition(
 	return scope;
 }
 
+export function createAnimationScope(
+	transitionName: string,
+	animations: Record<string, TransitionAnimationPair>
+) {
+	const hash = Math.random().toString(36).slice(2, 8);
+	const scope = `astro-${hash}`;
+	const sheet = new ViewTransitionStyleSheet(scope, transitionName);
+
+	addPairs(animations, sheet);
+
+	const styles = `<style>${sheet.toString()}</style>`;
```

Comment:
I would probably not wrap it in the tag and just pass the raw styles back and let them do what they want with them. With the tags you wouldn't be able to just do `{styles}` because it would be escaped, would have to do `<Fragment set:html={styles} />` which feels a little worse to me than `<style is:inline>{styles}</style>`

Reply:
ha, copied and pasted without giving it much thought. I have to admit that the result looks more natural without style tags.  Nevertheless, `<style is:inline>{styles}</style>` did not work for me. Are there any special spells required here?

Reply:
Ah, looks like you need to do: `<style set:text={styles}></style>`. I'm not sure if not being able to pass an expression as a child is intentional or a bug. 

In any event, I think having the raw styles and then having the flexibility to do it a couple of different ways is still better. Curious what you think.

Reply:
The API is definitely a big improvement now over what I had before. And I'm getting closer :)  `<style set:text={styles} />` now works for me after I removed all quotes from the stylesheet (since set:text escapes quotes). Removing the quotes seems safe to me. It only affects the scope-id, "new" and "old".